std/string.d

```diff
@@ -7385,25 +7385,27 @@ return it typed as a UTF string.
 `ubyte` becomes `char`, `ushort` becomes `wchar` and `uint`
 becomes `dchar`. Type qualifiers are preserved.
 
-When compiled with debug mode, this function performs an extra check to make
-sure the return value is a valid Unicode string.
-
 Params:
     arr = array of bytes, ubytes, shorts, ushorts, ints, or uints
 
 Returns:
     arr retyped as an array of chars, wchars, or dchars
 
+Throws:
+    `AssertError`, when the result is not a well-formed UTF string.
+
 See_Also: $(LREF representation)
 */
-auto assumeUTF(T)(T[] arr) pure
+auto assumeUTF(T)(T[] arr) pure nothrow
 if (staticIndexOf!(Unqual!T, ubyte, ushort, uint) != -1)
 {
     import std.traits : ModifyTypePreservingTQ;
+    import std.exception : assumeWontThrow;
     import std.utf : validate;
+
     alias ToUTFType(U) = AliasSeq!(char, wchar, dchar)[U.sizeof / 2];
-    auto asUTF = cast(ModifyTypePreservingTQ!(ToUTFType, T)[])arr;
-    debug validate(asUTF);
+    auto asUTF = cast(ModifyTypePreservingTQ!(ToUTFType, T)[]) arr;
+    assumeWontThrow(validate(asUTF));
```

Comment:
`assumeWontThrow` produces the following message which hides the real issue:
```
assumeWontThrow failed: Expression did throw
```

The assert should probably contain the exceptions message:


```suggestion
    scope ex = collectException!UTFException(validate(asUTF));
    assert(!ex, ex.msg);
```